File: Java version/app/src/main/java/it/unical/mat/parsers/asp/ASPSolversParser.java
package it.unical.mat.parsers.asp;

import it.unical.mat.parsers.asp.clingo.ClingoParserBaseVisitorImplementation;
import it.unical.mat.parsers.asp.dlv.DLVParserBaseVisitorImplementation;
import it.unical.mat.parsers.asp.dlv2.DLV2ParserBaseVisitorImplementation;
import it.unical.mat.parsers.asp.dlvhex.DLVHEXParserBaseVisitorImplementation;

public final class ASPSolversParser {
	private ASPSolversParser() {
		
	}
	
	public static void parseClingo(final ASPDataCollection answerSets, final String atomsList, final boolean two_stageParsing) {
		ClingoParserBaseVisitorImplementation.parse(answerSets, atomsList, two_stageParsing);
	}
	
	public static void parseDLV(final ASPDataCollection answerSets, final String atomsList, final boolean two_stageParsing) {
    	DLVParserBaseVisitorImplementation.parse(answerSets, atomsList, two_stageParsing);
	}
    
    public static void parseDLV2(final ASPDataCollection answerSets, final String atomsList, final boolean two_stageParsing) {
    	DLV2ParserBaseVisitorImplementation.parse(answerSets, atomsList, two_stageParsing);
    }
    
    public static void parseDLVHEX(final ASPDataCollection answerSets, final String atomsList, final boolean two_stageParsing) {
    	DLVHEXParserBaseVisitorImplementation.parse(answerSets, atomsList, two_stageParsing);
    }
}

File: CSharp version/it/unical/mat/parsers/asp/ASPSolversParser.cs
using it.unical.mat.parsers.asp.clingo;
using it.unical.mat.parsers.asp.dlv;
using it.unical.mat.parsers.asp.dlv2;
using it.unical.mat.parsers.asp.dlvhex;

namespace it.unical.mat.parsers.asp
{
    public static class ASPSolversParser
    {
        public static void ParseClingo(IASPDataCollection answerSets, string atomsList, bool two_stageParsing)
        {
            ClingoParserBaseVisitorImplementation.Parse(answerSets, atomsList, two_stageParsing);
        }

        public static void ParseDLV(IASPDataCollection answerSets, string atomsList, bool two_stageParsing)
        {
            DLVParserBaseVisitorImplementation.Parse(answerSets, atomsList, two_stageParsing);
        }

        public static void ParseDLV2(IASPDataCollection answerSets, string atomsList, bool two_stageParsing)
        {
            DLV2ParserBaseVisitorImplementation.Parse(answerSets, atomsList, two_stageParsing);
        }

        public static void ParseDLVHEX(IASPDataCollection answerSets, string atomsList, bool two_stageParsing)
        {
            DLVHEXParserBaseVisitorImplementation.Parse(answerSets, atomsList, two_stageParsing);
        }
    }
}